r2-core/src/main/java/com/linkedin/r2/transport/http/client/TimeoutCallback.java

```diff
@@ -67,13 +68,30 @@ public TimeoutCallback(ScheduledExecutorService executor, long timeout, TimeUnit
    * @param timeoutUnit the {@link TimeUnit} for the timeout parameter.
    * @param callback the {@link Callback} to be invoked on success or error.
    * @param timeoutThrowable the custom exception that will be used during the timeout
+   * @Deprecated use {@link #TimeoutCallback(ScheduledExecutorService, long, TimeUnit, Callback, Supplier)} instead
    */
+  @Deprecated
   public TimeoutCallback(ScheduledExecutorService executor, long timeout, TimeUnit timeoutUnit,
       final Callback<T> callback, final Throwable timeoutThrowable)
   {
     _timeout = new SingleTimeout<>(executor, timeout, timeoutUnit, callback, (callbackIfTimeout) -> callbackIfTimeout.onError(timeoutThrowable));
   }
 
+  /**
+   * Construct a new instance.
+   *
+   * @param executor the {@link ScheduledExecutorService} used to schedule the timeout
+   * @param timeout the timeout delay, in the specified {@link TimeUnit}.
+   * @param timeoutUnit the {@link TimeUnit} for the timeout parameter.
+   * @param callback the {@link Callback} to be invoked on success or error.
+   * @param timeoutThrowableSupplier the custom exception supplier that will be used during the timeout
+   */
+  public TimeoutCallback(ScheduledExecutorService executor, long timeout, TimeUnit timeoutUnit,
+      final Callback<T> callback, final Supplier<Throwable> timeoutThrowableSupplier)
+  {
+    _timeout = new SingleTimeout<>(executor, timeout, timeoutUnit, callback, callbackIfTimeout -> callbackIfTimeout.onError(timeoutThrowableSupplier));
```

Comment:
The default method introduced in Callback.java can be avoided if we invoke as follows
`callbackIfTimeout -> callbackIfTimeout.onError(timeoutThrowableSupplier.get())`

Reply:
Agree. See my earlier comments.